turtle/src/triple_allocator.rs

```diff
@@ -31,6 +31,14 @@ impl TripleAllocator {
         }
     }
 
+    pub fn complete_len(&self) -> usize {
+        self.complete_len
+    }
+
+    pub fn incomplete_len(&self) -> usize {
+        self.complete_len
```

Comment:
Maybe `incomplete_len`?

Reply:
ouch! and I'm using this in multiple `debug_assert` and that didn't break ?!?

Reply:
it didn't break because my assertions were wrong! But I now understand why, and I fixed them :)